src/SmartFormat/Extensions/DictionarySource.cs

```diff
@@ -25,28 +33,119 @@ public override bool TryEvaluateSelector(ISelectorInfo selectorInfo)
 
         var selector = selectorInfo.SelectorText;
 
-        // See if current is an IDictionary and contains the selector:
+        // See if current is an IDictionary (including generic dictionaries) and contains the selector:
         if (current is IDictionary rawDict)
             foreach (DictionaryEntry entry in rawDict)
             {
                 var key = entry.Key as string ?? entry.Key.ToString()!;
 
-                if (key.Equals(selector, selectorInfo.FormatDetails.Settings.GetCaseSensitivityComparison()))
-                {
-                    selectorInfo.Result = entry.Value;
-                    return true;
-                }
+                if (!key.Equals(selector, selectorInfo.FormatDetails.Settings.GetCaseSensitivityComparison()))
+                    continue;
+
+                selectorInfo.Result = entry.Value;
+                return true;
             }
 
-        // this check is for dynamics and generic dictionaries
-        if (current is not IDictionary<string, object?> dict) return false;
+        // This check is for dynamics (ExpandoObject):
+        if (current is IDictionary<string, object?> dict)
+        {
+            // We're using the CaseSensitivityType of the dictionary,
+            // not the one from Settings.GetCaseSensitivityComparison().
+            // This is faster and has less GC than Key.Equals(...)
+            if (!dict.TryGetValue(selector, out var val)) return false;
+
+            selectorInfo.Result = val;
+            return true;
+        }
+        
+        // This is for IReadOnlyDictionary<,> using Reflection
+        if (IsIReadOnlyDictionaryEnabled && TryGetDictionaryValue(current, selector,
```

Comment:
Making it optional makes sense, nice idea.

Reply:
Renamed to `IsIReadOnlyDictionarySupported`